### PR TITLE
ref(search): More clarity cleanup in grammar

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Many of the spaces tokens were not required. It should be much more clear now where spaces are needed.

- free text now has been more consistently broken into `free_text_unquoted` and `free_text_quoted`, which makes up the `free_text` token.

- Consistent naming for date / time filters

- Match ordering of time filters

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -92,12 +99,18 @@
 boolean_filter = negation? search_key sep boolean_value
 
 # aggregate numeric filter
-aggregate_filter          = negation? aggregate_key sep operator? (duration_format / numeric_value / percentage_format)
-aggregate_date_filter     = negation? aggregate_key sep operator? iso_8601_date_format
-aggregate_rel_date_filter = negation? aggregate_key sep operator? rel_date_format
+aggregate_filter = negation? aggregate_key sep operator? (duration_format / numeric_value / percentage_format)
+
+# aggregate for dates
+aggregate_time_filter = negation? aggregate_key sep operator? iso_8601_date_format
```
Comment (wedamija) [inline src/sentry/api/event_search.py:105]: Why the change from date -> time (reads weird when we have `iso_8601_date_format`)

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -356,29 +369,42 @@ def is_percentage_key(self, key):
...
+    def visit_boolean_term(self, node, children):
+        if not self.allow_boolean:
+            raise InvalidSearchQuery(
+                'Boolean statements containing "OR" or "AND" are not supported in this search'
+            )
+
+        children = self.flatten(self.remove_space(children))
+        return children[0].text.upper()
+
+    def visit_paran_term(self, node, children):
```
Comment (wedamija) [inline src/sentry/api/event_search.py:383]: `visit_paren_term`

Comment (EvanPurkhiser) [reply src/sentry/api/event_search.py:383]: damn does this do nothing then lol

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/api/event_search.py:99]: since the other ones were called time. But I could switch them all over to date.

File: src/sentry/api/event_search.py [hunk 3]
```diff
@@ -37,28 +37,35 @@
 
 event_search_grammar = Grammar(
     r"""
-search = (boolean_operator / paren_term / search_term)*
+search = spaces term*
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/api/event_search.py:40]: yeah, actually I'll inline it in the `term` token since we reuse that in some places where the space is required.

File: src/sentry/api/event_search.py (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/api/event_search.py:383]: anyway this is gonna go away :)

File: src/sentry/api/event_search.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/api/event_search.py:40]: Cool good suggestion, you can see `term` now has a `space`. I opted to add it there since we also use it in the paren_group.

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/api/event_search.py]: Okay changed everything to `date`.

File: src/sentry/api/event_search.py (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/api/event_search.py]: irrelevant now that `paren_term` went away.

**Review by wedamija — APPROVED** [review]
